### ISSUE TITLE
CAL-425: In (open)SUSE autologin is configured not for separate displaymanagers, but in /etc/sysconfig

### ISSUE THREAD

**Issue by calamares-user** [calamares/calamares#582]
Jira issue originally created by user embar-:

In openSUSE and SUSE Linux systems, changing variables for user autologin in displaymanagers' configuarations (sddm, kdm, gdm...) will not take effect.
In openSUSE and SUSE Linux systems, instead, display managers reads DISPLAYMANAGER_AUTOLOGIN variable from /etc/sysconfig/displaymanager configuration file.
Thus it would be nice to have separate case for (open)SUSE systems in Calamares displaymanagers module.

**Comment by vinyanalista** [comment]
How openSUSE users should fill the `displaymanagers` setting in [`displaymanager.conf`](https://github.com/calamares/calamares/blob/v3.2.11/src/modules/displaymanager/displaymanager.conf#L7)?

The distribution itself does:

```yaml
displaymanagers:
    None

sysconfigSetup: true
```

(I cropped just the relevant parts)

Source: [Open Build Service](https://build.opensuse.org/package/view_file/openSUSE:Leap:15.1/calamares/3.1.2-configuring_autologin_in_sysconfig.patch?expand=1)

I'm working on an openSUSE derivative and deploying Calamares as installer (by the way, thanks for the great work!).

Calamares fails after filling the filesystem:

![VirtualBox_Live_02_09_2019_23_01_48](https://user-images.githubusercontent.com/1576506/64139792-80db5a80-cdd8-11e9-90a1-201c0d93c0cb.png)

```
Boost.Python error in job "displaymanager".
<class 'AttributeError'>
'str' object has no attribute 'append'

Traceback:
File "/usr/lib64/calamares/modules/displaymanager/main.py", line 761, in run
    dm_names.append("sysconfig")
```

Where it fails: [`displaymanager/main.py`, line 761](https://github.com/calamares/calamares/blob/v3.2.11/src/modules/displaymanager/main.py#L761)

I'm actually not using the package provided by the distribution, but [a fork](https://build.opensuse.org/package/show/home:kamarada:15.1:dev/calamares) updated to v3.2.11, but I haven't changed `displaymanager.conf`. They are the same.

Here is all of it:

```yaml
# Configure one or more display managers (e.g. SDDM)
# with a "best effort" approach.
---
#The DM module attempts to set up all the DMs found in this list, in that precise order.
#It also sets up autologin, if the feature is enabled in globalstorage.
#The displaymanagers list can also be set in globalstorage, and in that case it overrides anything set up here.
displaymanagers:
    None

#Enable the following settings to force a desktop environment in your displaymanager configuration file:
#defaultDesktopEnvironment:
#    executable: "startkde"
#    desktopFile: "plasma"

#If true, try to ensure that the user, group, /var directory etc. for the
#display manager are set up correctly. This is normally done by the distribution
#packages, and best left to them. Therefore, it is disabled by default.
basicSetup: false

#If true, setup autologin for openSUSE. This only makes sense on openSUSE
#derivatives or other systems where /etc/sysconfig/displaymanager exists.
sysconfigSetup: true
```

Also, I enabled `displaymanager` in [`settings.conf`](https://github.com/kamarada/branding/blob/8da8b89954/calamares/settings.conf#L118)

Any ideas on what is going wrong and/or what is the fix?

**Comment by philmmanjaro** [comment]
Seems the [original approach](https://github.com/calamares/calamares/commit/9561bac1c8e323699fd2085066193b139afc6a5f) got updated due to the [code changes](https://github.com/calamares/calamares/commit/ebae7e1f476a28bd57b7d83d175ad7e156b875e1). @adriaandegroot what do you think?

**Comment by embar-** [comment]
This bug was indeed solved in 2017 for Calamares 3.1.2.
However regression made in 2018 by https://github.com/calamares/calamares/commit/ebae7e1f476a28bd57b7d83d175ad7e156b875e1 for calamares 3.2(.2), see https://github.com/calamares/calamares/issues/1253

Thus please close this old bug here and continue regression bug as https://github.com/calamares/calamares/issues/1253

**Comment by adriaandegroot** [comment]
@vinyanalista *None* is not a valid configuration value, although Python thinks it's fine and treats it as a string and **most** list methods work on it -- except `.append()` -- and so previously, while invalid, it meant that the module would check for DMs named "N", "o", "n" and "e" before going on with the sysconfig approach.

What will work right now is ["none"] which **is** a list, and which contains one element that won't be recognised as a DM, and then the sysconfig approach will be used. That works, but looks silly. Really what **should** be the solution is just to list "sysconfig" amongst the DMs.

But we'll continue the discussion over at #1253
